src/os/kqueue.rs

```diff
@@ -233,23 +239,23 @@ impl QueueableSealed for Signal {
 }
 impl Queueable for Signal {}
 
-/// Wait for a child process to exit.
+/// Wait for a process to exit.
 ///
 /// When registered into [`Async`](crate::Async) via [`with_filter`](AsyncKqueueExt::with_filter),
 /// it will return a [`readable`](crate::Async::readable) event when the child process exits.
 #[derive(Debug)]
-pub struct Exit(Option<Child>);
+pub struct Exit(NonZeroI32);
 
 impl Exit {
     /// Create a new `Exit` object.
-    pub fn new(child: Child) -> Self {
-        Self(Some(child))
+    pub fn new(pid: NonZeroI32) -> Self {
```

Comment:
I'd like to avoid a breaking change, can we make this a new method instead?

Reply:
Good point, a new `from_pid` constructor added in https://github.com/smol-rs/async-io/pull/207/commits/430846412d260388f5375f4e2aec94aa907444c6 instead. I made it `unsafe` to propagate the safety statement when creating a polling process from a pid.

Wdyt?

Reply:
I don't think it actually needs to be unsafe? While it's unsafe for FDs, PIDs are allowed to be racy IIRC

Reply:
The main reason behind the `unsafe` was to propagate the "safety" statement [defined in the polling process](https://github.com/smol-rs/polling/blob/master/src/os/kqueue.rs#L219) (and used by the reactor):

```rust
/// Create a `Process` from a PID.
    ///
    /// # Safety
    ///
    /// The PID must be tied to an actual child process.
    pub unsafe fn from_pid(pid: std::num::NonZeroI32, ops: ProcessOps) -> Self {
        Self {
            pid: unsafe { rustix::process::Pid::from_raw_unchecked(pid.get()) },
            ops,
            _lt: PhantomData,
        }
    }
```

With the new Exit status API, the struct can be crated for any given u32 value, thus the possibility of not being tied to an actual child process (differently than child.id()). This is the error when the provided pid is not attached to a process:
```rust
Error: Os { code: 3, kind: Uncategorized, message: "No such process" }
```

Reply:
Given that with the new `from_pid` it lets the user provide any PID value, do you still think the unsafe guidance is not needed?

Thanks for your feedback.